app/pages/signup/signup.js

```diff
@@ -60,15 +60,25 @@ var Signup = React.createClass({
     ];
   },
 
-  componentWillMount: function() {
-    var that = this;
-    if ((this.props.inviteKey || this.props.inviteKey === '' ) && this.props.checkInviteKey) {
-      this.props.checkInviteKey(this.props.inviteKey , function(valid) {
-        that.setState({loading: false, showWaitList: !valid });
+  isWaitListed:function(){
+
+    var hasInviteKey = !_.isEmpty(this.props.inviteKey) || this.props.inviteKey === '';
```

Comment:
if you don't set the key in the URL param then you get and empty string. Then the check in checkInviteKey() that is passed in will see if there is a config value for the expected key - which might not be set so then in that case and empty string is legit.

It was a workaround so you could still develop localhost without having to pass through keys all the time.